sdk/common.go

```diff
@@ -54,18 +57,27 @@ func JSONWithoutHTMLEncode(t interface{}) ([]byte, error) {
 }
 
 // FileMd5sum returns the md5sum ofr a file
-func FileMd5sum(filePath string) (md5sum string, err error) {
-	file, err := os.Open(filePath)
-	if err != nil {
-		return
+func FileMd5sum(file *os.File) (string, error) {
+	reader := bufio.NewReader(file)
+	hash := md5.New()
+	if _, err := io.Copy(hash, reader); err != nil {
+		return "", fmt.Errorf("FileMd5sum> error: %v", err)
```

Comment:
since you wrap the error everywhere this func is callled, I think the error should be "unable to copy file content to md5: %v"

something like "myFunction>" is usefull in log but not in an error.
go lint checks error string:
https://github.com/golang/lint/blob/85993ffd0a6cd043291f3f63d45d656d97b165bd/lint.go#L1208